src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/LazyLoadingValueHolderGenerator.php

```diff
@@ -38,4 +88,19 @@ public function generate(\ReflectionClass $originalClass, ClassGenerator $classG
             $destructor->setBody($newBody);
         }
     }
+
+    public function getProxifiedClass(Definition $definition): ?string
+    {
+        if (!$definition->hasTag('proxy')) {
+            return \class_exists($definition->getClass()) ? $definition->getClass() : null;
+        }
+        if (!$definition->isLazy()) {
+            throw new \InvalidArgumentException(sprintf('Invalid definition for service of class "%s": setting the "proxy" tag on a service requires it to be "lazy".', $definition->getClass()));
+        }
+        if (!isset($definition->getTag('proxy')[0]['interface'])) {
```

Comment:
what if we have multiple tags ? Should we support implementing multiple interfaces ?

Reply:
That's not supported by ProxyManager AFAIK (ping @Ocramius)
Should we throw?

Reply:
No, implementing multiple interfaces is not supported. I tried doing so when playing around with DCI implementations, but never pursued, because the use-case is too narrow compared with the added complexity.

Reply:
thanks for the confirmation, this situation is now throwing

Reply:
I managed to make it work actually, now supports proxying several interfaces :)
When they're not compatible, a PHP fatal error is thrown, there no other way around.